plugins/weather/wunderground_

```diff
@@ -0,0 +1,356 @@
+#!/bin/sh -eu
+# -*- sh -*-
+
+: << =cut
+
+=head1 NAME
+
+wunderground - Plugin to monitor weather stations through Weather Underground
+
+The precipitation rate is recalculated from the daily precipitation, as a
```

Comment:
This explanation is a bit hard to understand for me. It sounds like you want to work around a deficiency of the sensor's output. And additionally you want to improve the readability of the numbers.

Are you sure, that there is no cleaner way? Maybe even storing an SI base unit value - i.e. based on seconds?

Reply:
Essentially, rather than (well, in addition to) reading the current precipitation as a GAUGE, which will give incorrect aggregated data if we miss some samples, the idea is to use the cumulative daily precipitation as a DERIVE (same as using the total number of bytes on a network interface to derive the current throughput). The tradeoff is that instead of wrong aggregates on the longer-period graphs, you might get spikes on the daily graph, after some data loss. Any data loss that lasts more than a day, though, is completely lost.

The problem is that DERIVE only accepts integers, and the precipitation rate is in mm/hour. It felt a bit silly to use any smaller unit and try to pretend everything was fine, so I went the way of arbitrarily rescaling the number to the precision I wanted (so, 10 µm), and being verbose about it (which you found, so it worked!)

I can definitely rephrase the comment. I'm also not sure whether having those two similar metrics is actually needed, but time will tell. They have been in agreement with each other so far on my systems.

Reply:
OK - thanks for the explanation.

Since I do not feel qualified to judge, which approach would be most useful (or expected by the users), I leave it up to you. If you think, it is fine as it is, then feel free to keep it.

Reply:
Well, I'm not sure, so I'd rather leave both for now (;